frontend/src/main/scala/bloop/Cli.scala

```diff
@@ -89,8 +89,16 @@ object Cli {
       else parse(args, nailgunOptions)
     }
 
-    val exitStatus = run(cmd, NailgunPool(ngContext))
-    ngContext.exit(exitStatus.code)
+    try {
+      val exitStatus = run(cmd, NailgunPool(ngContext))
+      ngContext.exit(exitStatus.code)
+    } catch {
+      case x: java.util.concurrent.ExecutionException =>
+        // print stack trace of fatal errors thrown in asynchronous code, see https://stackoverflow.com/questions/17265022/what-is-a-boxed-error-in-scala
+        // the stack trace is somehow propagated all the way to the client when printing this
+        x.getCause.printStackTrace()
```

Comment:
Maybe?
```suggestion
        x.getCause.printStackTrace(ngContext.out)
```
This should probably also show up correctly via BSP

Reply:
Yeah, makes sense. Let's try